zilliqa/src/message.rs

```diff
@@ -454,8 +474,10 @@ impl Block {
             .flat_map(|v| v.public_key.as_bytes())
             .collect();
 
+        // Could the hash be all zeroes for genesis perhaps?
```

Comment:
I guess this might not work too well with shards in some way(?)

Reply:
It used to be all zeroes but was changed to be the real hash to avoid special case handling - iirc it was done first so node joining and block syncing are simpler, but it's very useful for shards too.
And conceptually, I do think it's cleaner for genesis to have a proper hash just like any block.